Automattic/wp-calypso
# Import: update goBack when coming from migration plugin flow

Related to #77492 

## Proposed Changes

* Currently if users come from the migration plugin flow and clicks back button during the flow, they will be navigate to `/import` step which ask them to fill in the address again. Ideally when users coming from the plugin flow, they will mostly change their target site instead of the source site. So this PR points users to `sitePicker` step if they're from plugin flow.
* We also fix some small state bugs include:
     * Adding both source site and target site as a queryKey to `migration-enabled` endpoint so it won't just depend on the source site because during the migration user might go back and change their target site.
     * Adding a `resetImportReceived` action to reset `isImporterStatusHydrated ` when user clicks `goBack`. Otherwise when it comes to the step again `isImporterStatusHydrated` is already true before fetching and you'll see a Page not found component.

## Testing Instructions



* Create a `JN site withe the Move to WordPress.com` plugin and connect to your WordPress.com account.
* Once you land on the plans page or content only page, click the back button on the left top, and see if it brings you to the `sitePicker` step
* Select a site and see if it brings you to the migration ready or plans page correctly without any glitches.

## Pre-merge Checklist
